ui/learn/src/map/mapSide.ts

```diff
@@ -89,19 +90,76 @@ function renderHome(ctrl: SideCtrl) {
         },
       }),
     ]),
-    m('div.actions', [
+    m(
+      'div.actions',
       progress > 0
-        ? m(
-            'a.confirm',
-            {
-              onclick: function () {
-                if (confirm(ctrl.trans.noarg('youWillLoseAllYourProgress'))) ctrl.reset();
+        ? [
+            m(
+              'label',
+              {
+                for: 'reset-categ-dropdown',
               },
-            },
-            ctrl.trans.noarg('resetMyProgress')
-          )
-        : null,
-    ]),
+              ctrl.trans.noarg('resetMyProgress')
+            ),
+            m(
+              'select',
+              {
+                name: 'reset-categ-dropdown',
+                id: 'reset-categ-dropdown',
+                oninput: (value: Event | null) => {
+                  const target = value?.target as HTMLSelectElement;
+                  const question = target.selectedOptions[0].getAttribute('question');
+                  if (question) {
+                    if (target.value !== '' && confirm(question)) {
+                      if (target.value === 'ALL') {
+                        ctrl.reset();
+                      } else {
+                        ctrl.partial_reset(target.value);
+                      }
+                    }
+                    target.value = '';
+                  }
+                },
+              },
+              ...[
+                ...[
+                  m(
```

Comment:
Can't you just remove these `...[` and the closing bracket? Just directly pass the parameters. And then just the `...stages.categs.map(..)` as the last "argument".

Reply:
I tried (to do without the first "...") and strangely couldn't.